c10/core/StorageImpl.cpp

```diff
@@ -36,4 +36,39 @@ StorageImplCreateHelper GetStorageImplCreate(DeviceType t) {
   return StorageImplCreate[device_type];
 }
 
+c10::intrusive_ptr<c10::StorageImpl> make_storage_impl(
+    c10::StorageImpl::use_byte_size_t use_byte_size,
+    c10::SymInt size_bytes,
+    c10::DataPtr data_ptr,
+    c10::Allocator* allocator,
+    bool resizable,
+    c10::optional<at::Device> device_opt) {
```

Comment:
Do I read correctly that both the DataPtr argument and this optional Device were added here?
Given that DataPtr has a device attribute already and a few of the call site don't have it, why not just have an optional DataPtr?

Worst case, the allocator can definitely tell you which device it allocates for

Reply:
> Do I read correctly that both the DataPtr argument and this optional Device were added here? Given that DataPtr has a device attribute already and a few of the call site don't have it, why not just have an optional DataPtr?
> 
> Worst case, the allocator can definitely tell you which device it allocates for

Same to ezyang's suggestion, here DataPtr is allowed to be nullptr so will not be an optional variable.